packages/core/test/invoke.test.ts

```diff
@@ -1601,12 +1539,104 @@ describe('invoke', () => {
     });
 
     it('should treat a callback source as an event stream', (done) => {
+      const intervalMachine = createMachine<{
+        interval: number;
+        count: number;
+      }>({
+        id: 'interval',
+        initial: 'counting',
+        context: {
+          interval: 10,
+          count: 0
+        },
+        states: {
+          counting: {
+            invoke: {
+              id: 'intervalService',
+              src: (ctx) =>
+                fromCallback((cb) => {
+                  const ivl = setInterval(() => {
+                    cb({ type: 'INC' });
+                  }, ctx.interval);
+
+                  return () => clearInterval(ivl);
+                })
+            },
+            always: {
+              target: 'finished',
+              guard: (ctx) => ctx.count === 3
+            },
+            on: {
+              INC: { actions: assign({ count: (ctx) => ctx.count + 1 }) },
+              SKIP: 'wait'
+            }
+          },
+          wait: {
+            on: {
+              // this should never be called if interval service is properly disposed
+              INC: { actions: assign({ count: (ctx) => ctx.count + 1 }) }
+            },
+            after: {
+              50: 'finished'
+            }
+          },
+          finished: {
+            type: 'final'
+          }
+        }
+      });
       interpret(intervalMachine)
         .onDone(() => done())
         .start();
     });
 
     it('should dispose of the callback (if disposal function provided)', (done) => {
```

Comment:
This test could be simplified like this:
```ts
    it('should dispose of the callback (if disposal function provided)', () => {
      const spy = jest.fn();
      const intervalMachine = createMachine({
        id: 'interval',
        initial: 'counting',
        states: {
          counting: {
            invoke: {
              id: 'intervalService',
              src: () => fromCallback(() => spy)
            },
            on: {
              NEXT: 'idle'
            }
          },
          idle: {}
        }
      });
      const actorRef = interpret(intervalMachine).start();

      actorRef.send({ type: 'NEXT' });

      expect(spy).toHaveBeenCalled();
    });
```

(I've realized that this was just moved after already writing this)